app/scripts/lib/xhr.js

```diff
@@ -67,6 +67,7 @@ define([
         // make sure to set the dataType for Firefox <21. See issue #1930
         dataType: 'json',
         type: options.type,
+        timeout: options.timeout || 0,
```

Comment:
Are there any side effects of setting the timeout to `0`? Does the xhr request never timeout, or does it timeout immediately?